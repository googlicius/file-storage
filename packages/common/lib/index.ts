export function common() {
  // TODO
  return 'this is common';
}
