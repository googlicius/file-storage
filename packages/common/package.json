{
  "name": "@file-storage/common",
  "version": "1.2.7",
  "description": "Common utilities, types, interfaces,... for `file-storage`.",
  "author": "Dang Nguyen <haidang009@gmail.com>",
  "homepage": "",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/googlicius/file-storage.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "scripts": {
    "build": "yarn clean && yarn compile",
    "clean": "rimraf -rf ./dist",
    "compile": "tsc -p tsconfig.build.json"
  },
  "dependencies": {
    "sharp": "^0.28.3"
  },
  "devDependencies": {
    "@types/sharp": "^0.28.5",
    "rimraf": "~3.0.2",
    "typescript": "~4.3.5"
  }
}
