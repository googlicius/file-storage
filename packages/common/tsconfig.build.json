{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "outDir": "./dist"
  },
  "include": ["src/**/*.ts"],
  "exclude": ["**/*.spec.ts"]
}
