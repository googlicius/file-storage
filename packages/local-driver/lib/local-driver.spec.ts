import fs from 'fs';
import Storage from '@file-storage/core';
import { DriverName, FileNotFoundError, getRootCwd } from '@file-storage/common';

describe('Local Disk', () => {
  beforeAll(() => {
    Storage.config({
      diskConfigs: [
        {
          driver: DriverName.LOCAL,
          name: 'local',
          root: 'storage',
        },
      ],
    });
  });

  test('Default disk is local', () => {
    expect(Storage.name).toEqual('local');
  });

  test('Upload image to local success', () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    return expect(
      Storage.disk('local').put(fileReadStream, 'test_upload/bird.jpeg'),
    ).resolves.toMatchObject({
      success: true,
      message: 'Uploading success!',
    });
  });

  test('Upload image to local success (Using default disk)', () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    return expect(Storage.put(fileReadStream, 'test_upload/bird$$.jpeg')).resolves.toMatchObject({
      success: true,
      message: 'Uploading success!',
    });
  });

  test('Upload local large image will uploaded to many formats', () => {
    const imageFileStream = fs.createReadStream(
      getRootCwd() + '/test/support/images/photo-1000x750.jpeg',
    );
    return expect(
      Storage.put(imageFileStream, 'my-photo/photo-1000x750.jpeg'),
    ).resolves.toMatchObject({
      success: true,
      message: 'Uploading success!',
      formats: {
        thumbnail: {
          name: 'thumbnail_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 208,
          height: 156,
          size: 15.9,
          path: 'my-photo/thumbnail_photo-1000x750.jpeg',
        },
        large: {
          name: 'large_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 1000,
          height: 750,
          size: 184.49,
          path: 'my-photo/large_photo-1000x750.jpeg',
        },
        medium: {
          name: 'medium_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 750,
          height: 562,
          size: 105.12,
          path: 'my-photo/medium_photo-1000x750.jpeg',
        },
        small: {
          name: 'small_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 500,
          height: 375,
          size: 52.33,
          path: 'my-photo/small_photo-1000x750.jpeg',
        },
      },
    });
  });

  test('Upload image from URI to local', () => {
    return expect(
      Storage.disk('local').uploadImageFromExternalUri(
        'https://4.img-dpreview.com/files/p/E~TS590x0~articles/3925134721/0266554465.jpeg',
        'test_upload/test_image_from_uri.jpeg',
      ),
    ).resolves.toMatchObject({
      success: true,
      message: 'Uploading success!',
    });
  });

  test('Upload image from URI to local (Using default disk)', () => {
    return expect(
      Storage.uploadImageFromExternalUri(
        'https://4.img-dpreview.com/files/p/E~TS590x0~articles/3925134721/0266554465.jpeg',
        'test_upload/test_image_from_uri.jpeg',
      ),
    ).resolves.toMatchObject({
      success: true,
      message: 'Uploading success!',
    });
  });

  test('Upload a file is not an image from URI to local', () => {
    const uri = 'https://previews.magnoliabox.com/kew/mb_hero/kewgar066/MUS-FAPC1114_850.jpg';
    return expect(
      Storage.disk('local').uploadImageFromExternalUri(uri, 'test_upload/test_image_from_uri.jpeg'),
    ).rejects.toEqual(new Error('Not an image: ' + uri));
  });

  test('Delete image from local (Using default disk)', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    const filePath = 'test_upload/image123.jpeg';
    await Storage.put(fileReadStream, filePath);

    return expect(Storage.delete(filePath)).resolves.toBeTruthy();
  });

  test('Delete file does not exists', () => {
    return expect(Storage.delete('not-exists.jpeg')).rejects.toThrowError(FileNotFoundError);
  });

  test('Download image from local', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    // Storage.put
    await Storage.put(fileReadStream, 'test_upload/bird2.jpeg');
    return expect(Storage.get('test_upload/bird2.jpeg')).resolves.toBeTruthy();
  });

  test('Download not exists image from local error', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.put(fileReadStream, 'test_upload/bird2.jpeg');
    return expect(Storage.get('not-exist.jpeg')).rejects.toThrowError(FileNotFoundError);
  });

  test('File is exists', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.put(fileReadStream, 'bird-images/bird.jpeg');

    return expect(Storage.exists('bird-images/bird.jpeg')).resolves.toEqual(true);
  });

  test('File is not exists', async () => {
    return expect(Storage.exists('not-exists.jpeg')).resolves.toEqual(false);
  });

  test('Get file size', async () => {
    const fileReadStream2 = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('local').put(fileReadStream2, 'bird-images/bird.jpeg');

    return expect(Storage.size('bird-images/bird.jpeg')).resolves.toEqual(56199);
  });

  test('Last modified', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('local').put(fileReadStream, 'bird-images/bird.jpeg');
    const lastMod = await Storage.lastModified('bird-images/bird.jpeg');
    expect(typeof lastMod).toBe('number');
  });
});
