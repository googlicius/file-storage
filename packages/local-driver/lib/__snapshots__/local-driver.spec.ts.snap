// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local Disk Upload image from URI to local 1`] = `
{
  "message": "Uploading success!",
  "success": true,
}
`;

exports[`Local Disk Upload local large image will uploaded to many formats 1`] = `
{
  "formats": {
    "large": {
      "ext": "jpeg",
      "hash": null,
      "height": 750,
      "mime": "jpeg",
      "name": "large_photo-1000x750.jpeg",
      "path": "my-photo/large_photo-1000x750.jpeg",
      "size": 184.49,
      "width": 1000,
    },
    "medium": {
      "ext": "jpeg",
      "hash": null,
      "height": 562,
      "mime": "jpeg",
      "name": "medium_photo-1000x750.jpeg",
      "path": "my-photo/medium_photo-1000x750.jpeg",
      "size": 107.23,
      "width": 750,
    },
    "small": {
      "ext": "jpeg",
      "hash": null,
      "height": 375,
      "mime": "jpeg",
      "name": "small_photo-1000x750.jpeg",
      "path": "my-photo/small_photo-1000x750.jpeg",
      "size": 52.4,
      "width": 500,
    },
    "thumbnail": {
      "ext": "jpeg",
      "hash": null,
      "height": 156,
      "mime": "jpeg",
      "name": "thumbnail_photo-1000x750.jpeg",
      "path": "my-photo/thumbnail_photo-1000x750.jpeg",
      "size": 16.14,
      "width": 208,
    },
  },
  "message": "Uploading success!",
  "name": "photo-1000x750.jpeg",
  "path": "my-photo/photo-1000x750.jpeg",
  "success": true,
}
`;

exports[`Local Disk append should append a text to a file 1`] = `
"First line
Appended text"
`;
